File: 0001~0500/0389. Find the Difference/main.py
"""
Source: https://leetcode.com/problems/find-the-difference/
Date: 2023/2/16
Skill:
Runtime: 30 ms, faster than 91.8%
Memory Usage: 13.9 MB, less than 22.34%
Time complexity:
Space complexity:
Constraints:

"""

import math
from typing import List, Tuple, Optional
from collections import defaultdict, deque
from heapq import heapify, heappush, heappop, nsmallest, heappushpop
import bisect
import heapq
import functools


class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        res = 0
        for c in s:
            res ^= ord(c)
        for c in t:
            res ^= ord(c)

        return chr(res)


if __name__ == "__main__":
    s = Solution

File: 0001~0500/0389. Find the Difference/main.cpp
/**
 * Source: t.ly/Cw7hQ
 * Date: 2023/9/25
 * Skill:
 * Runtime: 0 ms, faster than 100.00% of C++ online submissions
 * Memory Usage: 6.97 MB, less than 16.46% of C++ online submissions
 * Time complexity: O(n)
 * Space complexity: O(n)
 * Constraints:
 *
 *
 */

#include <vector>
#include <iostream>
#include <algorithm>
#include <map>
#include <set>
#include <unordered_set>
#include <queue>
#include <numeric>
#include <tuple>
#include <stack>
#include <bitset>

using namespace std;

#define ll long long
typedef pair<int, int> pairs;
typedef unsigned long long ULL;
using PULL = pair<ULL, ULL>;


class Solution {
public:
    char findTheDifference(string s, string t) {
        for (int i = 0; i < s.length(); i++)
            t[i + 1] += t[i] - s[i];
        return t[s.length()];
    }
};



int main() {
    Solution s;
    char res = s.findTheDifference("", "y");
    cout << res << endl;
    vector<vector<int>> arr1{{5, 3},
                             {2, 5},
                             {7, 4},
                             {10, 3}};
    vector<vector<int>> arr2{{0, 1},
                             {2, 2},
                             {1, 4}};
    string word = "aaabccccacacacaabcbaaabacbbbcabcbcaacbabccbababcabaacaacbbcbaabc";
}